socs/agents/lakeshore425/agent.py

```diff
@@ -123,25 +124,46 @@ def acq(self, session, params):
 
             last_release = time.time()
             while self.take_data:
-                if time.time() - last_release > 1.:
-                    last_release = time.time()
-                    if not self.lock.release_and_acquire(timeout=10):
-                        self.log.warn(f"Failed to re-acquire lock, currently held by {self.lock.job}.")
+                if self.dev is None:
+                    try:
+                        self.dev = ls.LakeShore425(self.port)
+                        self.log.info('Connected to LS425')
+                    except (ConnectionRefusedError, serial.serialutil.SerialException):
+                        self.log.error(
+                            "Could not connect to LS425. "
+                            "Retrying after 30 sec..."
+                        )
+                        time.sleep(30)
                         continue
-
-                Bfield = self.dev.get_field()
-                current_time = time.time()
-                data = {
-                    'timestamp': current_time,
-                    'block_name': 'mag_field',
-                    'data': {'Bfield': Bfield}
-                }
-
-                self.agent.publish_to_feed('mag_field', data)
-                session.data.update({'timestamp': current_time})
-                self.agent.feeds['mag_field'].flush_buffer()
-
-                time.sleep(sleep_time)
+                try:
+                    if time.time() - last_release > 1.:
+                        last_release = time.time()
+                        if not self.lock.release_and_acquire(timeout=10):
+                            self.log.warn(f"Failed to re-acquire lock, currently held by {self.lock.job}.")
+                            continue
+
+                    Bfield = self.dev.get_field()
+                    current_time = time.time()
+                    data = {
+                        'timestamp': current_time,
+                        'block_name': 'mag_field',
+                        'data': {'Bfield': Bfield}
+                    }
+
+                    self.agent.publish_to_feed('mag_field', data)
+                    session.data.update({'timestamp': current_time})
+                    self.agent.feeds['mag_field'].flush_buffer()
+
+                    session.degraded = False
+                    time.sleep(sleep_time)
+                except Exception:
+                    self.log.error(
+                        "Decive reports readiness to read but returned no data. "
+                        "Reconnect to LS425."
+                    )
```

Comment:
Can we narrow the exception handling to a specific (few) exception(s) here? It looks like you're targeting a specific error.

The communication errors you're trying to handle look like they'd only really come in on line 145, with that `self.dev.get_field()`.

The general `except Exception` is fine to also have, but we should also make it clear that the exception caught was unexpected and log what it is -- like we do [here in the tcp module](https://github.com/simonsobs/socs/blob/b7295afc8f6399a6f9a5762a1f51992fe19e1d11/socs/tcp.py#L47-L49).

Reply:
That makes sense I made it more specific